src/python/turicreate/toolkits/activity_classifier/_activity_classifier.py

```diff
@@ -233,27 +262,82 @@ def create(dataset, session_id, target, features=None, prediction_window=100,
 
     # Always create MXNet models, as the pred_model is later saved to the state
     # If MPS is used - the loss_model will be overwritten
-    loss_model, pred_model = _define_model_mxnet(len(target_map), prediction_window,
-                                                 predictions_in_chunk, context)
+    loss_model, pred_model = _define_model_mxnet(len(target_map), prediction_window, predictions_in_chunk, context)
 
     if use_mps:
         loss_model = _define_model_mps(batch_size, len(features), len(target_map),
                                        prediction_window, predictions_in_chunk, is_prediction_model=False)
 
         log = _fit_model_mps(loss_model, data_iter, valid_iter, max_iterations, verbose)
-
     else:
-        # Train the model using Mxnet
-        log = _fit_model_mxnet(loss_model, data_iter, valid_iter,
-                         max_iterations, num_mxnet_gpus, verbose)
+        from .._mxnet import mxnet as _mx
+        # Initialize the weights in MXNet and then pass them in TensorFlow
+        dummy_data_iter = _SFrameSequenceIter(chunked_data, len(features),
+                                    prediction_window, predictions_in_chunk,
+                                    batch_size, use_target=use_target, mx_output=True)
+        loss_model.bind(data_shapes=dummy_data_iter.provide_data, label_shapes=dummy_data_iter.provide_label)
+        loss_model.init_params(initializer=_mx.init.Xavier())
+        net_params, aux = loss_model.get_params()
+
+        # Dictionary with layer names as key and initialised value as value
+        net_params.update(aux)
+
+        if params['use_tensorflow']:
+            ac_model = ActivityTensorFlowModel(net_params, batch_size, len(features), len(target_map), prediction_window, predictions_in_chunk)
+            # Train the model using Tensorflow
+            log = _fit_model_tf(ac_model, net_params, data_iter, valid_iter, max_iterations, verbose, 1e-3)
+        else:
+            # Train the model using Mxnet
+            log = _fit_model_mxnet(loss_model, dummy_data_iter, valid_iter, max_iterations, num_mxnet_gpus, verbose)
 
     # Set up prediction model
     pred_model.bind(data_shapes=data_iter.provide_data, label_shapes=None,
                     for_training=False)
-
+    
     if use_mps:
         mps_params = loss_model.export()
         arg_params, aux_params = _ac_weights_mps_to_mxnet(mps_params, _net_params['lstm_h'])
+
+    elif params['use_tensorflow']:
+        # Copy the weights back in the MXNet format
```

Comment:
Let's extract this code into a utility function appearing in your TF-specific file